awscli/compat.py

```diff
@@ -68,10 +70,15 @@ def get_stdout_text_writer():
         # unicode to the returned stream.  Note that get_stdout_text_writer
         # just returns sys.stdout in the PY3 section above because python3
         # handles this.
-        return codecs.getwriter(locale.getpreferredencoding())(sys.stdout)
+        encoding = locale.getpreferredencoding()
+        if encoding == '':
+            encoding = locale.getpreferredencoding(do_setlocale=False)
+        return codecs.getwriter(encoding)(sys.stdout)
 
     def compat_open(filename, mode='r', encoding=None):
         # See docstring for compat_open in the PY3 section above.
         if 'b' not in mode:
-            encoding = locale.getpreferredencoding()
+            encoding = locale.getpreferredencoding(do_setlocale=False)
```

Comment:
I actually did change this back lase June, but never pushed it.  It's up there now.